apps/api/src/auth/services/api-key/api-key.service.ts

```diff
@@ -0,0 +1,64 @@
+import { singleton } from "tsyringe";
+
+import { ApiKeyInput, ApiKeyOutput, ApiKeyRepository } from "@src/auth/repositories/api-key/api-key.repository";
+import { AuthService } from "@src/auth/services/auth.service";
+import { ApiKeyGeneratorService } from "./api-key-generator.service";
+
+@singleton()
+export class ApiKeyService {
+  constructor(
+    private readonly apiKeyRepository: ApiKeyRepository,
+    private readonly authService: AuthService,
+    private readonly apiKeyGenerator: ApiKeyGeneratorService
+  ) {}
+
+  async findAll(): Promise<ApiKeyOutput[]> {
+    return await this.apiKeyRepository.accessibleBy(this.authService.ability, "read").find({ userId: this.authService.currentUser.id });
+  }
+
+  async findById(id: string): Promise<ApiKeyOutput | undefined> {
+    const key = await this.apiKeyRepository.accessibleBy(this.authService.ability, "read").findOneBy({ id, userId: this.authService.currentUser.id });
+
+    if (!key) return undefined;
```

Comment:
throw 404?

Reply:
this is how we do it in other places. REST is all about unification after all

Reply:
We throw a 404 in the controller if the key is not found

Reply:
Ah, ok, for some reason i was thinking i'm looking at the controller ;)